### PR TITLE
Kma 160777391 display move dates service

### PR DESCRIPTION
**Pull request by kimallen**
## Description

This inserts the proper date values in the dates summary when a date is clicked or when there is an existing requested_pickup_date.

## Reviewer Notes
~Note that the dates are currently not correct because the api retrieving the dates is incorrect.  When that api is [fixed](https://www.pivotaltracker.com/story/show/161032441) the dates will be correct.~ Dates are now correct.  There is another story to change how dates are displayed if there are dates that straddle a weekend [here](https://www.pivotaltracker.com/story/show/161105619)

## Setup
Go to service member hhg flow dates selection page.  Click on a date.  the dates summary should appear.

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] There are no aXe warnings for UI. (don't know- aXe was having issues)
* [ ] This works in IE.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/160777391) for this change

## Screenshots
capture date selection

### CODE REVIEW

File: src/scenes/Moves/Hhg/api.js [hunk 1]
```diff
@@ -9,3 +10,13 @@ export async function GetShipment(moveId, shipmentId) {
...
+
+export async function GetMoveDatesSummary(moveId, moveDate) {
+  const client = await getClient();
+  const response = await client.apis.moves.showMoveDatesSummary({
+    moveId,
+    move_date: formatDateString(moveDate),
+  });
+  checkResponse(response, 'failed to get hhg shipment due to server error');
+  return response.body;
+}
```
Comment (reggieriser) [inline src/scenes/Moves/Hhg/api.js]: I'm guessing you accidentally left this in after your first pass at the "old" way.  Is this still used anywhere?

File: src/shared/Swagger/request.js [hunk 2]
```diff
@@ -126,6 +123,14 @@ export function swaggerRequest(getClient, operationPath, params, options = {}) {
...
         }
 
+        if (schemaKey.indexOf('Payload') !== -1) {
+          const newSchemaKey = schemaKey.replace('Payload', '');
+          console.warn(
+            `Using 'Payload' as a response type prefix is deprecated. Please rename ${schemaKey} to ${newSchemaKey}`,
+          );
+          schemaKey = newSchemaKey;
+        }
+
```
Comment (reggieriser) [inline src/shared/Swagger/request.js]: Nice!

File: src/scenes/Moves/Hhg/DatesSummary.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        <tbody>
+          <tr>
+            <th colSpan="2">Preferred Moving Dates Summary</th>
+          </tr>
+          <tr>
+            <td>Movers Packing</td>
+            <td>
+              {this.formatDate(packDates[0])} -{' '}
+              {this.formatDate(packDates[packDates.length - 1])}*
+              <span className="estimate">(estimated)</span>
```
Comment (reggieriser) [inline src/scenes/Moves/Hhg/DatesSummary.jsx]: What if there is just one day of packing?  Won't we get a range still?  Same thing with transit below.

File: src/shared/Entities/schema.js [hunk 4]
```diff
@@ -77,3 +77,10 @@ export const availableMoveDates = new schema.Entity(
   {},
   { idAttribute: 'start_date' },
 );
+
+// MoveDatesSummary
+export const moveDatesSummary = new schema.Entity(
+  'moveDatesSummaries',
+  {},
+  { idAttribute: summary => summary.pickup[0] },
+);
```
Comment (reggieriser) [inline src/shared/Entities/schema.js]: The endpoint already has an `id` field on it which was a combination of the move ID and the move (pickup) date.  Is there a reason we're not using it here?

File: src/scenes/Moves/Hhg/DatesSummary.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  render() {
+    const { moveDates } = this.props;
+    const pickupDates = get(moveDates, 'pickup', []);
+    const packDates = get(moveDates, 'pack', []);
+    const deliveryDates = get(moveDates, 'delivery', []);
+    const transitDates = get(moveDates, 'transit', []);
+    const reportDates = get(moveDates, 'report', []);
+
+    return (
+      <table>
```
Comment (reggieriser) [inline src/scenes/Moves/Hhg/DatesSummary.jsx]: It'd be nice to have some "Loading" component here if we don't have the moving dates yet.  When I click a new date, I get a half second or so of ranges with no dates (just the hyphen).

**Comment by reggieriser** [comment]
FYI, I created a new user and went through the flow and did not have the above problem (presumably because no shipment record existed).  But once I selected a move date and went to the next page (causing a shipment record to be created with the selected date) and then came back -- the problem noted above started happening.

File: src/scenes/Moves/Hhg/api.js (hunk 1 above)
Comment (kimallen) [reply src/scenes/Moves/Hhg/api.js]: Nope, you're right.  good catch.

File: src/shared/Swagger/request.js (hunk 2 above)
Comment (kimallen) [reply src/shared/Swagger/request.js]: Can't claim that- Jim gave it to me while pairing

File: src/scenes/Moves/Hhg/DatesSummary.jsx (hunk 3 above)
Comment (kimallen) [reply src/scenes/Moves/Hhg/DatesSummary.jsx]: fixed

**Comment by kimallen** [comment]
Thanks for the helpful code review!  I fixed the bug you mentioned when clicking on the dates.

**Comment by reggieriser** [comment]
Changes look good.  I created a new SM and tested creating a new shipment, then also going back and testing the flow when the SM had already picked a date.  Both worked fine.

One side note ... I first tried using an existing SM in my local database that was there from a previous generation of e2e data.  That SM's move happened to have the same origin and destination, so the endpoint returned a 500 response code (and a log message that it `Could not find transit time for 10500 lbs and 0 miles`).  However, the UI just got stuck on the "Loading" message.  Not sure if we have a standard way of presenting 500-type errors in the UI.

Also, I see there's a failing e2e test which sounds like something I believe you saw before: `CypressError: Timed out retrying: Expected to find content: 'Movers Packing' but never did.`  I get that locally too.  I'm going ahead and approving the PR from my perspective while you sort that one out.

**Pull request merged**